src/device/mod.rs

```diff
@@ -1043,6 +1045,12 @@ impl Runtime {
     async fn set_private_key(&mut self, private_key: &SecretKey) -> Result {
         // TODO: create a global controll state to consolidate all entities
 
+        let should_validate_keys = self.features.validate_keys.0;
+        let meshnet_is_on = self.requested_state.meshnet_config.is_some();
+        if should_validate_keys && meshnet_is_on {
```

Comment:
This is probably not too relevant, but if the `set_private_key` is called with the key, which is already set, probably no error should be thrown, as the main point of the validation is to ensure, that public key in meshmap and private key in WG are from the same pair. This would throw error even if they are of the same pair.

Reply:
Other than this, +2.0

Reply:
I skipped adding that at first because it doesn't make sense to try to set the same key again, but you are right. It's quite the unnecessary error to return, so I fixed it